spec/index.bs

```diff
@@ -12606,6 +12579,22 @@ Issue: Write normative text about timestamp value resets.
 
 Issue: Because timestamp query provides high-resolution GPU timestamp, we need to decide what constraints, if any, are on its availability.
 
+<div algorithm class=validusage>
+    <dfn abstract-op>Validate timestampWrites</dfn>({{GPUDevice}} |device|,
+    <code>({{GPUComputePassTimestampWrites}} or {{GPURenderPassTimestampWrites}})</code> |timestampWrites|)
+
+    Return `true` if the following requirements are met, and `false` if not.
+
+    - If |timestampWrites| is not [=list/is empty|empty=],
+        {{GPUFeatureName/"timestamp-query"}} must be [=enabled for=] |device|.
+    - No two entries in |timestampWrites| may have the same `location`.
+    - No two entries in |timestampWrites| may have the same `querySet` and `queryIndex` pair.
```

Comment:
The same `querySet` and `queryIndex` pair is allowed on compute pass, this should only be validated on render pass.

Reply:
Well spotted! Thanks for reviewing, will fix soon.

Reply:
FWIW I think #3808 _will_ make this obsolete, but the goal here with this PR is not to change anything.
Moved this validation to the render pass only.